packages/eslint-plugin/src/rules/no-extra-parens.ts

```diff
@@ -141,7 +141,18 @@ export default util.createRule<Options, MessageIds>({
       },
       BinaryExpression: binaryExp,
       CallExpression: callExp,
-      // ClassDeclaration
+      ClassDeclaration(node) {
+        if (node.superClass?.type === AST_NODE_TYPES.TSAsExpression) {
+          return rules.ClassDeclaration({
+            ...node,
+            superClass: {
+              ...node.superClass,
+              type: AST_NODE_TYPES.SequenceExpression as any,
+            },
+          });
+        }
+        return rules.ClassDeclaration(node);
+      },
       // ClassExpression
```

Comment:
Could you add tests for `const _ = class { ... }` class **expressions** too? I suspect they'll also need the same fix as **declarations**.

Reply:
Hi, @JoshuaKGoldberg   I added test cases and handled class expressions. Thanks for the review